pkg/model/awsmodel/utils.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsmodel
+
+import (
+	"strings"
+
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awstasks"
+	"k8s.io/kops/upup/pkg/fi/utils"
+)
+
+// IsPrefixListID checks whether the string resembles a prefix list ID
+func IsPrefixListID(id string) bool {
+	return strings.HasPrefix(id, "pl-")
+}
+
+func LimitIngress(t *awstasks.SecurityGroupRule, cidr string) *awstasks.SecurityGroupRule {
```

Comment:
Perhaps make this a receiver on `awstasks.SecurityGroupRule`? Perhaps call it `SetPrefix()`?

Reply:
I've added that as `SetCidrOrPrefix`, as I thought that `SetPrefix` wasn't fully covering the functionality.

Reply:
My thinking was that a CIDR specifies a prefix. A "pl-" string is a prefix *list*. Both CIDR and PrefixList specify the subset of the address space that the SGR applies to.

I suppose "Prefix" is incorrectly singular when applied to prefix lists. I'm trying to find a good term to express the concept of the subset of address space.

Reply:
Yes agreed, basically either limit the ingress to certain address(blocks).

So that's why I initially chose LimitIngress

Reply:
Except a security group rule can be for egress instead.